Orange/data/io.py

```diff
@@ -379,16 +379,24 @@ def write(cls, filename, data):
 
     @classmethod
     def write_table_metadata(cls, filename, data):
-        if isinstance(filename, str) and getattr(data, 'attributes', None):
+
+        def write_file(fn):
             if all(isinstance(key, str) and isinstance(value, str)
                    for key, value in data.attributes.items()):
-                with open(filename + '.metadata', 'w', encoding='utf-8') as f:
+                with open(fn, 'w', encoding='utf-8') as f:
                     f.write("\n".join("{}: {}".format(*kv)
                                       for kv in data.attributes.items()))
             else:
-                with open(filename + '.metadata', 'wb') as f:
+                with open(fn, 'wb') as f:
                     pickle.dump(data.attributes, f, pickle.HIGHEST_PROTOCOL)
 
+        if isinstance(filename, str):
+            metafile = filename + '.metadata'
+            if path.exists(metafile) and not getattr(data, 'attributes', None):
+                unlink(metafile)
```

Comment:
I confess that I didn't know what `unlink` does. If it's not too dear to you, would you change it to `remove`? :) [Python documentation for unlink](https://docs.python.org/3/library/os.html#os.unlink) just redirects the reader to `remove`.